### PR TITLE
Add predictors.auto_configure to automatically configure predictors for data sources

### PR DESCRIPTION
**Pull request by maxhutch**
# Citrine Python PR

## Description 
Please briefly explain the goal of the changes/this PR.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [ ] I have bumped the version in setup.py

### CODE REVIEW

File: src/citrine/resources/predictor.py [hunk 1]
```diff
@@ -62,3 +63,33 @@ def check_for_update(self, predictor_id: UUID) -> Optional[Predictor]:
...
+        This method will return an unregistered predictor generated by inspecting the
+        training data and attempting to automatically configure the predictor.
+        The automatic configuration tries to be as inclusive as possible.
+
+        Parameters
+        ----------
+        training_data: DataSource
+            The data to configure the predictor to model.
+        pattern: str
+            The predictor pattern to use, either "PLAIN" or "FORMULATION"
```
Comment (latture) [inline src/citrine/resources/predictor.py:79]: Do we have user docs somewhere that explain the difference between the two options?

**Review by latture — APPROVED** [review]
It might be nice to add documentation on what is configured for the default predictor and the differences between the two options at some point. However, the code looks good, and these docstrings are fine for alpha functionality.

File: src/citrine/resources/predictor.py [hunk 2]
```diff
@@ -62,3 +63,33 @@ def check_for_update(self, predictor_id: UUID) -> Optional[Predictor]:
...
+
+        Returns
+        -------
+        Predictor
+            Automatically configured predictor for the training data
+
+        """
+        if pattern not in {"FORMULATION", "PLAIN"}:
+            msg = "Expected pattern: {}.  Expected 'FORMULATION' or 'PLAIN'.".format(pattern)
```
Comment (latture) [inline src/citrine/resources/predictor.py:88]: Did you mean for this message to start with `Expected pattern` or is this a typo?

Comment (maxhutch) [reply src/citrine/resources/predictor.py]: Typo

**Review by latture — APPROVED** [review]

**Pull request merged**